### DISCUSSION TITLE
cronicle behind Cloudflare ZeroTrust

### DISCUSSION THREAD

**Discussion by nycalex** [jhuckaby/Cronicle#726] [Q&A] [answered]
hello
i am setting up chronicle behind Cloudflare Zero Trust.  

install is on Debian 12.
nodejs v21

For clarity - this is not the Cloudflare distributed CDN cache, it's a different cloudflare product for remote access (kinda like VPN replacement)

I assign a temp fqdn like scheduler.mydomain.com in Cloudflare.
in the setup json i configure base app url to match scheduler.mydomain.com
local fqdn hostname is different.

after i authenticate to scheduler.mydomain.com, i am tunneled to Cronicle.

I do get a page, first time it said something about finding master (don't remember wording)

but after that all i have is a Cronicle logo and a rectengular box. Page tab is called 'Loading..."

No progress after waiting a 20 min.

i'm watching hstat, ram or cpu are not busy, plenty of disk space too.

what am i missing??? 
Thanks

**Comment by jhuckaby** [comment]
Need to check the logs, both on the client side (open the browser dev tools and check the console), and the server-side where Cronicle is running.

Cronicle requires a WebSocket connection from the browser to the back-end server, so I suspect this is where your problem is.  I don't know anything about Cloudflare Zero trust, so my advice is limited here, but the websocket connection is going to be the key -- I'm almost sure of it.

If CloudFlare doesn't allow WebSockets, you might try: https://github.com/jhuckaby/Cronicle/blob/master/docs/Configuration.md#socket_io_transports

Also, make sure you have your server hostname registered properly in Cronicle's database.  Cronicle is extremely sensitive about server hostnames.  See: https://github.com/jhuckaby/Cronicle/wiki/Troubleshooting#server-hostnames

**Comment by nycalex** [comment, marked as answer]
i got it to work. two things were needed.

while i'm still accessing it as scheduler.mydomain.com, i set the base_app_url in config.json  to match the linux hostname where cronicle is running. 

second thing - in ZeroTrust config, under Tunnels, in the setting of public hostname (scheduler.mydomain.com), under additional settings, HTTP, i configured HTTP host header = linux hostname

[updated for clarity]  websockets had to be enabled too in cloudflare 

Thanks for helping!
